examples/text_classification/train.py

```diff
@@ -67,12 +62,15 @@ def train(lr_, num_epoch, data_):
         data_: the data used to train the model
     """
 
-    data = DataLoader(data_, batch_size=batch_size, shuffle=True,
-                      collate_fn=generate_batch, num_workers=args.num_workers)
     optimizer = torch.optim.SGD(model.parameters(), lr=lr_)
     scheduler = torch.optim.lr_scheduler.StepLR(optimizer, 1, gamma=args.lr_gamma)
-    num_lines = num_epochs * len(data)
+    num_lines = num_epochs * len(sub_train_) / batch_size * split_ratio
```

Comment:
why do you need to multiply this by the split_ratio and divide by the batch_size? The only thing that should have changed is the length of the dataset. Also, you're instantiating the DataLoader on each epoch, which can be a costly operation.

Reply:
Now that we've moved over to using StepLR I think we don't need this number anymore anyway, right?